src/Tests/Microsoft.NET.Publish.Tests/PublishItemsOutputGroupOutputsTests.cs

```diff
@@ -0,0 +1,111 @@
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using FluentAssertions;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.Commands;
+using Microsoft.NET.TestFramework.ProjectConstruction;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.NET.Publish.Tests
+{
+    public class PublishItemsOutputGroupOutputsTests : SdkTest
+    {
+        public PublishItemsOutputGroupOutputsTests(ITestOutputHelper log) : base(log)
+        {
+        }
+
+        private readonly static List<string> FrameworkAssemblies = new List<string>()
+        {
+            "api-ms-win-core-console-l1-1-0.dll",
+            "System.Runtime.dll",
+            "WindowsBase.dll",
+        };
+
+        [Theory]
+        [InlineData(true, false)]
+        [InlineData(true, true)]
+        [InlineData(false, false)]
+        public void RunPublishItemsOutputGroupOutputsTest(bool specifyRid, bool singleFile)
+        {
+            var testProject = new TestProject()
+            {
+                Name = "TestPublishItemsOutputGroupOutputs",
+                TargetFrameworks = "netcoreapp3.0",
+                IsExe = true
+            };
+
+            testProject.AdditionalProperties["RuntimeIdentifiers"] = "win-x86";
+            testProject.AdditionalProperties["RestorePackagesPath"] = @"$(MSBuildProjectDirectory)\..\pkg";
+            if (specifyRid)
+            {
+                testProject.AdditionalProperties["RuntimeIdentifier"] = "win-x86";
+            }
+
+            if (singleFile)
+            {
+                testProject.AdditionalProperties["PublishSingleFile"] = "true";
+            }
+
+            var testAsset = _testAssetsManager.CreateTestProject(testProject);
+
+            var restoreCommand = new RestoreCommand(testAsset);
+            restoreCommand
+                .Execute()
+                .Should()
+                .Pass();
+
+            var command = new GetValuesCommand(
+                Log,
+                Path.Combine(testAsset.Path, testProject.Name),
+                testProject.TargetFrameworks,
+                "PublishItemsOutputGroupOutputs",
+                GetValuesCommand.ValueType.Item)
+            {
+                DependsOnTargets = "Publish",
+                MetadataNames = { "OutputPath" },
+            };
+
+            command.Execute().Should().Pass();
+            var items = from item in command.GetValuesWithMetadata()
+                        select new
+                        {
+                            Identity = item.value,
+                            OutputPath = Path.GetFileName(Path.GetFullPath(Path.Combine(testAsset.Path, testProject.Name, item.metadata["OutputPath"])))
```

Comment:
Why the call to `Path.GetFullPath` here if it's just going to strip off everything with the call to `Path.GetFileName`?

```suggestion
                            OutputPath = Path.GetFileName(item.metadata["OutputPath"])
```

Reply:
Directly call FullPath will put current PWD in front, which is wrong. It is more correct even for an intermediate value to combine the right path first.

Reply:
But the call to GetFileName is going to trim off all the directory information that was added either via the Path.Combine or Path.GetFullPath call.

Reply:
I know, but at one point the path points to a non exist file. It does not matter in the end but feels weird. I use GetFullPath to normalize the path from / to \\. GetFileName gives the wrong answer without it: GetFileName(c:\folder/file.dll) ->"folder/file.dll"